staging/src/k8s.io/kubectl/pkg/cmd/auth/whoami.go

```diff
@@ -142,45 +152,74 @@ func NewCmdWhoAmI(restClientGetter genericclioptions.RESTClientGetter, streams g
 var (
 	notEnabledErr = fmt.Errorf(
 		"the selfsubjectreviews API is not enabled in the cluster\n" +
-			"enable APISelfSubjectReview feature gate and authentication.k8s.io/v1alpha1 API")
+			"enable APISelfSubjectReview feature gate and authentication.k8s.io/v1alpha1 or authentication.k8s.io/v1beta1 API")
 
 	forbiddenErr = fmt.Errorf(
 		"the selfsubjectreviews API is not enabled in the cluster or you do not have permission to call it")
 )
 
 // Run prints all user attributes.
 func (o WhoAmIOptions) Run() error {
-	sar := &authenticationv1alpha1.SelfSubjectReview{}
-	response, err := o.authClient.SelfSubjectReviews().Create(context.TODO(), sar, metav1.CreateOptions{})
+	var (
+		res runtime.Object
+		err error
+	)
+
+	res, err = o.authV1beta1Client.
+		SelfSubjectReviews().
+		Create(context.TODO(), &authenticationv1beta1.SelfSubjectReview{}, metav1.CreateOptions{})
 	if err != nil {
-		switch {
-		case errors.IsForbidden(err):
-			return forbiddenErr
-		case errors.IsNotFound(err):
-			return notEnabledErr
-		default:
-			return err
+		if errors.IsNotFound(err) {
+			// Fallback to Alpha API if Beta is not enabled
+			res, err = o.authV1alpha1Client.
+				SelfSubjectReviews().
+				Create(context.TODO(), &authenticationv1alpha1.SelfSubjectReview{}, metav1.CreateOptions{})
+			if err != nil {
+				return checkEnabledOrForbidden(err)
+			}
+		} else {
+			return checkEnabledOrForbidden(err)
 		}
 	}
-	return o.resourcePrinterFunc(response, o.Out)
+	return o.resourcePrinterFunc(res, o.Out)
+}
+
+func checkEnabledOrForbidden(err error) error {
+	switch {
+	case errors.IsForbidden(err):
+		return forbiddenErr
+	case errors.IsNotFound(err):
+		return notEnabledErr
+	default:
+		return err
+	}
+}
+
+func getUserInfo(obj runtime.Object) (authenticationv1.UserInfo, error) {
+	switch obj.(type) {
+	case *authenticationv1alpha1.SelfSubjectReview:
+		return obj.(*authenticationv1alpha1.SelfSubjectReview).Status.UserInfo, nil
+	case *authenticationv1beta1.SelfSubjectReview:
+		return obj.(*authenticationv1beta1.SelfSubjectReview).Status.UserInfo, nil
+	default:
+		return authenticationv1.UserInfo{}, fmt.Errorf("object is not SelfSubjectReview")
```

Comment:
might be helpful to include the type for debugging if we hit this:

```suggestion
		return authenticationv1.UserInfo{}, fmt.Errorf("unexpected response type %T", obj)
```

Reply:
Applied this one.